mrtt-ui/src/components/ProjectAreaMap.js

```diff
@@ -0,0 +1,185 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import PropTypes from 'prop-types'
+import { Typography } from '@mui/material'
+import Map, {
+  NavigationControl,
+  FullscreenControl,
+  ScaleControl,
+  GeolocateControl
+} from 'react-map-gl'
+import { DropzoneArea } from 'react-mui-dropzone'
+import GeoPropTypes from 'geojson-prop-types'
+
+import language from '../language'
+import MapDrawControl, { drawControlRef } from './MapDrawControl'
+import MapDrawInfo from './MapDrawInfo'
+import handleFileLoadEvent from '../library/handleFileLoadEvent'
+
+import 'mapbox-gl/dist/mapbox-gl.css'
+import '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.css'
+
+const MAPBOX_TOKEN = process.env.REACT_APP_MAPBOX_ACCESS_TOKEN
+
+const acceptedFileMimeTypes = [
+  'application/json',
+  'application/geo+json',
+  'application/vnd.geo+json',
+  '.geojson',
+  'application/vnd.google-earth.kml+xml', // .kml
+  // Shapefile (zip)
+  'application/zip',
+  'application/octet-stream',
+  'application/x-zip-compressed',
+  'multipart/x-zip'
+]
+const lineTypes = ['Line', 'MultiLine', 'LineString']
+const pointTypes = ['Point', 'MultiPoint']
+const polygonTypes = ['Polygon', 'MultiPolygon']
+
+const getPolygonOnlyFeatureCollection = (geojsonFeatureCollection) => {
+  return {
+    ...geojsonFeatureCollection,
+    features: geojsonFeatureCollection.features.filter((feature) =>
+      polygonTypes.includes(feature.geometry.type)
+    )
+  }
+}
+
+const ProjectAreaMap = ({
+  height,
+  extent,
+  siteAreaFeatureCollection,
+  setSiteAreaFeatureCollection
+}) => {
+  const [uploadLineCount, setUploadLineCount] = useState(0)
+  const [uploadPointCount, setUploadPointCount] = useState(0)
+  const [error, setError] = useState()
+
+  // When extent changes
+  useEffect(() => {
+    if (!mapRef.current) return
+    const animationProps = { padding: 40, duration: 1000 }
+    if (extent) {
+      mapRef.current.fitBounds(
+        [
+          [extent[0], extent[1]],
+          [extent[2], extent[3]]
+        ],
+        animationProps
+      )
+    } else {
+      mapRef.current.zoomTo(0, animationProps)
+    }
+  }, [extent])
+
+  const onAddGeomFile = async (files) => {
+    if (drawControlRef) {
```

Comment:
You're right. I think they were separate because there was previously some other code that needed to run whether or not `files.length` was true. Updated